src/test/unit_tests/test_namespace_fs.js

```diff
@@ -335,13 +336,29 @@ mocha.describe('namespace_fs', function() {
             const read_data = read_res.join();
             assert.strictEqual(Buffer.compare(read_data, data), 0);
 
+            const read_res_part = buffer_utils.write_stream();
+            await ns_tmp.read_object_stream({
+                bucket: mpu_bkt,
+                key: mpu_key,
+                part_number: 1
+            }, dummy_object_sdk, read_res_part);
+            const part_data = read_res_part.join();
+            assert.strictEqual(Buffer.compare(part_data, data.subarray(0, part_size)), 0);
+
             const md = await ns_tmp.read_object_md({
                 bucket: upload_bkt,
                 key: mpu_key,
             }, dummy_object_sdk);
             console.log('read_object_md response', inspect(md));
             assert.deepStrictEqual(xattr, md.xattr);
 
+            const part_md = await ns_tmp.read_object_md({
```

Comment:
let's add more tests, these are some examples - 
1. 3 parts, all same size, get/head all and expect the same size but different data
2. 3 parts, last one is a different size, check the data, get/head all and expect different sizes, different data
3. 3 parts, each one different size, expect a failure